tests/calculators/test_qe.py

```diff
@@ -57,75 +60,99 @@ def qe_cmd_and_pseudo(tmp_path_factory):
-    kw, prop = qe_kpoints_and_kwargs(
-        Atoms("H", cell=[1, 1, 1], pbc=[True, False, False]),
-        dict(kpts=(2, 3, 4), koffset=True),
-        ["energy", "stress", "forces"],
-    )
+    atoms = Atoms("H", cell=[1, 1, 1], pbc=[True, False, False])
+    properties = ["energy", "stress", "forces"] 
+    kw["koffset"] = True
+    calc = wfl.calculators.espresso.Espresso(**kw)
+    calc.atoms = atoms.copy()
+    properties = calc.setup_params_for_this_calc(properties)
+
+    assert "tstress" in calc.parameters
+    assert not calc.parameters["tstress"]
+
+    assert "tprnfor" in calc.parameters
+    assert calc.parameters["tprnfor"]
 
-    assert "tstress" in kw
-    assert not kw["tstress"]
+    assert "stress" not in properties
 
-    assert "tprnfor" in kw
-    assert kw["tprnfor"]
+    assert calc.parameters["kpts"] == (2, 1, 1)
+    assert calc.parameters["kspacing"] is None
+    assert calc.parameters["koffset"] == (1, 0, 0)
 
-    assert "stress" not in prop
-    assert kw["kpts"] == (2, 1, 1)
-    assert kw["kspacing"] is None
-    assert kw["koffset"] == (1, 0, 0)
 
     # koffset in mixed PBC
-    kw, prop = qe_kpoints_and_kwargs(
-        Atoms("H", cell=[1, 1, 1], pbc=[True, False, False]),
-        dict(kpts=(2, 3, 4), koffset=False),
-        ["energy", "forces"],
-    )
-    assert kw["koffset"] is False
+    atoms = Atoms("H", cell=[1, 1, 1], pbc=[True, False, False])
+    properties = ["energy", "forces"] 
+    kw["koffset"] = False 
+    calc = wfl.calculators.espresso.Espresso(**kw)
+    calc.atoms = atoms.copy()
+    properties = calc.setup_params_for_this_calc(properties)
+
+    assert calc.parameters["koffset"] is False
+
 
     # PBC mixed -- kspacing
-    kw, prop = qe_kpoints_and_kwargs(
-        Atoms("H", cell=[1, 1, 1], pbc=[True, False, False]),
-        dict(kspacing=0.1, koffset=(0, 1, 0)),
-        ["energy", "stress", "forces"],
-    )
-    assert "tstress" in kw
-    assert not kw["tstress"]
+    atoms = Atoms("H", cell=[1, 1, 1], pbc=[True, False, False])
+    properties = ["energy", "stress", "forces"] 
+    kw["kspacing"] = 0.1
+    kw["koffset"] = (0, 1, 0)
+    calc = wfl.calculators.espresso.Espresso(**kw)
+    calc.atoms = atoms.copy()
+    properties = calc.setup_params_for_this_calc(properties)
+
+    assert "tstress" in calc.parameters
+    assert not calc.parameters["tstress"]
 
-    assert "tprnfor" in kw
-    assert kw["tprnfor"]
+    assert "tprnfor" in calc.parameters
+    assert calc.parameters["tprnfor"]
 
-    assert "stress" not in prop
-    assert kw["kpts"] == (63, 1, 1)
-    assert kw["kspacing"] is None
-    assert kw["koffset"] == (0, 0, 0)
+    assert "stress" not in properties
+    assert calc.parameters["kpts"] == (63, 1, 1)
+    assert calc.parameters["kspacing"] is None
+    assert calc.parameters["koffset"] == (0, 0, 0)
 
 
 @pytest.mark.xfail(reason="PP file changes. Even before that hard-wired values are wrong, also calculation does not converge with default conv_thr")
```

Comment:
What should we do with this test? Update new values to check against? Or remove this test and just check that the calculation is finished, as is done in `test_wfl_Espresso_calc`?

Reply:
I think it would be good to have at least one test that confirms that the actual results are correct.  IF another test already does this, no need for this one, but if there isn't, updating the potentials and reference values in this one would be better.

Reply:
done.